src/sentry_value.c

```diff
@@ -639,30 +640,16 @@ sentry__value_append_bounded(sentry_value_t value, sentry_value_t v, size_t max)
 
     list_t *l = thing->payload._ptr;
 
-    if (l->len < max) {
+    if (l->len <= max) {
```

Comment:
Since we now store [start_idx, breadcrumb 1, bc 2, ...] the full list will be max+1 long

Reply:
That is true, but can we also express it as you described it in the code?
```suggestion
    if (l->len < max + 1) {
```
This makes the intention much more evident.